sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -656,6 +656,48 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
     }
   }
 
+  /**
+   * Need to convert cudf unsigned integer to wider signed integer that parquet expects
+   * After spark 3.2.0, parquet read uint8 as int16, uint16 as int32, uint32 as int64
+   * TODO uint64 -> Decimal(20,0) depends CUDF, see issue #3475
+   *
+   * @param group the schema
+   * @return if has unsigned integer
+   */
+  def existsUnsignedType(group: GroupType): Boolean = {
+    group.getFields.asScala.exists(
+      field => {
+        if (field.isPrimitive) {
+          val t = field.getOriginalType
+          (t == OriginalType.UINT_8) || (t == OriginalType.UINT_16) || (t == OriginalType.UINT_32)
+        } else {
+          existsUnsignedType(field.asGroupType)
+        }
+      }
+    )
+  }
+
+  def needDecimalCast(cv: ColumnView, dt: DataType): Boolean = {
+    cv.getType.isDecimalType && !GpuColumnVector.getNonNestedRapidsType(dt).equals(cv.getType())
+  }
+
+  def needUnsignedToSignedCast(cv: ColumnView, dt: DataType): Boolean = {
+    (cv.getType.equals(DType.UINT8) && dt.isInstanceOf[ShortType]) ||
+      (cv.getType.equals(DType.UINT16) && dt.isInstanceOf[IntegerType]) ||
+      (cv.getType.equals(DType.UINT32) && dt.isInstanceOf[LongType])
+  }
+
+  def decimalCastOrUnsignedCast(cv: ColumnView, dt: DataType): ColumnView = {
```

Comment:
NIT: 
Better to add comments about when this will be called.
```suggestion
  private def decimalCastOrUnsignedCast(cv: ColumnView, dt: DataType): ColumnView = {
```